express-notes-api/index.js

```diff
@@ -0,0 +1,111 @@
+const fs = require('fs');
+const express = require('express');
+const app = express();
+const dataJson = require('./data.json');
+const notes = dataJson.notes;
+
+app.use(express.json());
+
+// request all notes:
+app.get('/api/notes', (req, res) => {
+  const notesArray = [];
+
+  for (const curNote in notes) {
+    notesArray.push(notes[curNote]);
+  }
+
+  res.status(200).json(notesArray);
+});
+
+// request a specific note:
+app.get('/api/notes/:id', (req, res) => {
+  const id = Number(req.params.id);
+  // verify id is positive integer and is included in the data:
+  if (id < 0 || isNaN(id)) {
+    res.status(400).json({ error: 'id must be a positive integer' });
+  } else if (id in notes === false) {
+    res.status(404).json({ error: `cannot find note with id ${id}` });
+  } else {
+    res.status(200).json(notes[id]);
+  }
+});
+
+// request to add a note:
+app.post('/api/notes', (req, res) => {
+  const newNote = req.body;
+
+  // verify that content is included:
+  if (Object.keys(newNote).length === 0) {
```

Comment:
This would allow for any key name to be used not just `content`. So here it would be better to directly check for the `content` property.
```JAVASCRIPT
if (!newNote.content) {
  // No content property set or has empty value, send error
}
```

Reply:
Absolutely, right, thanks!